src/shared/clients/s3Client.ts

```diff
@@ -301,9 +301,10 @@ export class DefaultS3Client {
      * Assigns the target content type based on the mime type of the file.
      * If content type cannot be determined, defaults to {@link DEFAULT_CONTENT_TYPE}.
      *
+     * @returns The S3.ManagedUpload stream
      * @throws Error if there is an error calling S3 or piping between streams.
      */
-    public async uploadFile(request: UploadFileRequest): Promise<void> {
+    public async uploadFile(request: UploadFileRequest): Promise<any> {
```

Comment:
Can we fix this type to return `Promise<S3.ManagedUpload>`?
If that causes issues then drop the `async` keyword

Reply:
fixed it